deploy/StableJoeStaking.js

```diff
@@ -38,22 +38,34 @@ module.exports = async function ({ getNamedAccounts, deployments }) {
         proxyContract: "OpenZeppelinTransparentProxy",
         execute: {
           init: {
-            methodName: "initializeSmolJoes",
-            args: [smolJoes],
+            methodName: "initialize",
+            args: [
+              rewardToken,
+              joeAddress,
+              feeCollector,
+              depositFeePercent,
+              smolJoes,
+            ],
           },
         },
       },
       log: true,
     });
     if (sJoe.newlyDeployed) {
       console.log("Initializing implementation for safe measure...");
-      const impl = await ethers.getContract("StableJoeStaking_Implementation");
-      await impl.initialize(
+      const sJoeImpl = await ethers.getContract(
+        "StableJoeStaking_Implementation"
+      );
+      await sJoeImpl.initialize(
         rewardToken,
         joeAddress,
         feeCollector,
-        depositFeePercent
+        depositFeePercent,
+        smolJoes
       );
+      console.log("Setting Smol Joes...");
+      const sJoeProxy = await ethers.getContract("StableJoeStaking");
+      await sJoeProxy.setSmolJoes(smolJoes);
```

Comment:
If it's newly deployed then this is not necessary, as it would set it with initialize

Reply:
On the upgrade though, it will be "newly deployed" but it won't call initialize.

Reply:
Oh indeed, my bad I thought it was setting it on the implementation again